### ISSUE TITLE
w1 ₃₂→₁₆ w2

### ISSUE THREAD

**Issue by mn200** [CakeML/cakeml#713]
Make it so.  Depends on #499

**Comment by mn200** [comment]
Actually, casts are unary, so perhaps we could have this as a suffix, so that
```
     w1 ₃₂→₁₆
```

would be the 16 bit value corresponding to `w1`.

Alternatively, and this is the one that Magnus prefers, have 
```
        ₃₂←₁₆ w2
```
which would be the 32 bit version of (16 bit value) `w2`.

**Comment by konrad-slind** [comment]
Possibly naive question. Why not have a general operation

   width(w,n)

which maps word w to its corresponding n-width version? Being binary, the
"width" operation could even be an infix, although I am not sure that you
are even considering concrete syntax.

**Comment by mn200** [comment]
You actually need to know both source and destination sizes, so in some sense it's a ternary operation.  Moreover, the source and dest numbers have to be literals; they're not really arguments  in the way that the `w` is.   

(This is all supposing that we have words of arbitrary widths floating around (as *per* issue #499).)

**Comment by konrad-slind** [comment]
Right. I was operating under the assumption that w would have a type from
which its size could be calculated. (I did this in a language we worked on
at Collins. Yes, the target width was a literal.)

**Comment by ordinarymath** [comment]
This issue can't be done until #499 is finished
